### ISSUE TITLE
cllient-side caching using http headers

### ISSUE THREAD

**Issue by bushaev** [paulcwarren/spring-content#54]
I was trying to make use of client side caching with Spring Content. One of the ways to do it with Spring Data Rest is by using `http` headers such as `ETag` or `Last-Modified`, which can be included in the response if corresponding java annotations are used ( such as `@Versioin` for `Etag` and  `@LastModifiedDate` for `Last-Modified`). 
Unfortunately, Spring content doesn't pick up on these annotations and does not return needed headers, more than that, using `@Version` annotation causes the exception being thrown while trying to upload file to entity:
```
org.springframework.web.util.NestedServletException: Request processing failed; nested exception is org.springframework.orm.ObjectOptimisticLockingFailureException: Object of class [com.package.ClassName] with identifier [59284]: optimistic locking failed; nested exception is org.hibernate.StaleObjectStateException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.package.ClassName#59284]
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:982)
	at org.springframework.web.servlet.FrameworkServlet.doPut(FrameworkServlet.java:888)
```
However, the file does upload.

So the question is, is it possible to include this kind of headers to the response without writing custom controller ?

**Comment by paulcwarren** [comment]
Hi @bushaev, yes this item has been on our todo list.  We first had to support `@Version` in some reasonable way which we have just finished doing actually.  So now optimistic locking is extended to any content associated with an Entity that is being `@Version`ed.  This was part of a wider "Locking and Versioning" story that also includes pessimisting locking (from a user perspective) and versioning of entities.  This is all sitting on master and will be released soon as part of `0.5.0`.  You could grab it if you want to get the latest SNAPSHOT though.  Anyways, we can now add support in Spring Content REST for `@Version` as well as `@LastModifiedDate`.   We'll pick this up and do the other bug fix at the same time.

**Comment by paulcwarren** [comment]
Hi @bushaev, we'll look into adding the http response headers as suggested shortly.  Just to let you know though.  The last set of commits on HEAD (0.6.0-SNAPSHOT) added support for `@Version` by extending the optimistic locking semantics to any Spring Content Resources associated with the `@Version`ed entity.
